src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -159,20 +160,33 @@ public struct TestResultSet
         /// </summary>
         private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
         {
+
+            var className = testNode.GetAttribute("classname");
+            var methodName = testNode.GetAttribute("methodname");
+           
+            // VS expected FullName to be the actual class+type name,optionally with parameter types
+            // in parenthesis, but they must fit the pattern of a value returned by object.GetType().
+            // It should _not_ include custom name or param values (just their types).
+            // However, the "fullname" from NUnit's file generation is the custom name of the test, so
+            // this code must convert from one to the other.
+            // Reference: https://github.com/microsoft/vstest-docs/blob/master/RFCs/0017-Managed-TestCase-Properties.md
+            var fullname = className + "." + methodName;
```

Comment:
Nit: It might be worth using a variable name other than `fullname` so that this isn't confused with `testNode.GetAttribute("fullname")`.